### PR TITLE
Adding Authentication to SFTT Frontend

### PR DESCRIPTION
**Pull request by jackblanc**

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import AxiosInstance from '../auth/axiosInstance';
+
+const CREATE_TEAM = '/api/v1/protected/teams';
+export const createTeam = async team => AxiosInstance.post(CREATE_TEAM, team);
```
Comment (bensenescu) [inline src/api/api.js:4]: The big downside to this pattern is that you probably want to try catch anything thats async

File: src/auth/axiosInstance.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  request.headers['X-Access-Token'] = tokenService.getAccessToken();
+  return request;
+});
+
+const INVALID_ACCESS_TOKEN = 'Given access token is expired or invalid';
+
+AxiosInstance.interceptors.response.use(
+  response => response,
+  (error) => {
+    const originalRequest = error.config;
```
Comment (bensenescu) [inline src/auth/axiosInstance.js:25]: What is error.config

File: src/api/api.js (hunk 1 above)
Comment (jackblanc) [reply src/api/api.js:4]: Doesn't the .then.catch accomplish the same thing? Pretty sure 
```
try { 
  // code1
} catch (e) {
  // code2
}
```
is equivalent to
```
.then((response) => code1).catch((error) => code2)
```

File: src/auth/axiosInstance.js (hunk 2 above)
Comment (jackblanc) [reply src/auth/axiosInstance.js:25]: The original request in its entirety

File: src/api/api.js (hunk 1 above)
Comment (liammoyn) [reply src/api/api.js:4]: Yep, the .then / .catch pattern is equivalent as long as we actually conform to that pattern in the calling code.

Comment (jackblanc) [reply src/api/api.js:4]: So which do we want? I prefer this by a ton haha

**Pull request merged**